pipeline_dp/pipeline_operations.py

```diff
@@ -349,3 +346,213 @@ def count_per_element(self, col, stage_name: typing.Optional[str] = None):
 
     def reduce_accumulators_per_key(self, col, stage_name: str = None):
         return self.map_values(self.group_by_key(col), accumulator.merge)
+
+
+class Sentinel(Enum):
+    EOI = "EndOfIterations"
+
+
+def _multiproc_iter_input(iter_obj):
+    yield from iter_obj
+    yield Sentinel.EOI
+
+
+def _multiproc_iter_output(iter_obj):
+    for item in iter_obj:
+        if item is Sentinel.EOI:
+            break
+        yield item
+
+
+# workaround for passing lambda functions to multiprocessing
+# according to https://medium.com/@yasufumy/python-multiprocessing-c6d54107dd55
+_pool_current_func = None
+
+
+def _pool_worker_init(func):
+    global _pool_current_func
+    _pool_current_func = func
+
+
+def _pool_worker(row):
+    if row is Sentinel.EOI:
+        return Sentinel.EOI
+    return _pool_current_func(row)
+
+
+class _LazyMultiProcIterator:
```

Comment:
Could you please add documentation for this class with a short explanation how it works. Something like, that paralellization is achived with using parallel map from multiprocessing.

Reply:
Done, hope to satisfaction.

Reply:
Thanks, it looks great!